src/renderer/tvgShape.h

```diff
@@ -209,10 +209,9 @@ struct Shape::Impl
 
     void strokeTrim(float begin, float end, bool simultaneous)
     {
-        if (!rs.stroke) {
-            if (begin == 0.0f && end == 1.0f) return;
-            rs.stroke = new RenderStroke();
-        }
+        if ((begin == 0.0f && end == 1.0f) || tvg::zero(begin - end + 1.0f)) return;
```

Comment:
I think this condition should be on line 213 as it is necessary to reset the value.

Reply:
please take a look at the sample
since trim is added as the last one, `rs.stroke` already exists, so this condition won't be checked and stroke won't be drawn, which is a bug

no matter whether stroke is created or not, if ```(begin == 0.0f && end == 1.0f) || tvg::zero(begin - end + 1.0f)``` there is no point to add trimming. Am I missing something?

Reply:
@mgrudzinska Please consider resetting the case. When the stroke trim has valid values, reset it to zero. Your change ignores it.

Reply:
@hermet thank you, I forgot about the case of overriding the trim.

Please take a look now. I am overriding the values provided by the user, which isn't optimal because if the user queries them with a getter, he gets different values than he provided. Do you think I should handle this differently?

Reply:
Although in many cases, the values are adjusted, and the getter will return something different than what the user provided so it might not be a problem.